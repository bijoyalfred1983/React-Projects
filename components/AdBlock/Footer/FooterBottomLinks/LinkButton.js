/* @flow */

import React from 'react';
import type { Element } from 'react';
import styles from './styles.scss';
import { jsTrack } from '../../../containers/Analytics/ClickTrack';

type Props = { bottomLinks: Array<Object> };

export const trackFooterLink = (e, linkName) => {
  jsTrack.trackFooterNavMenu(linkName);
};
const LinkButton = ({ bottomLinks }: Props): Element<'div'> => (
  <div className={styles.foooterBottomLinks}>
    <ul>
      {bottomLinks &&
        bottomLinks.map(linkbutton => (
          <li key={linkbutton.id}>
            <a
              href={linkbutton.url}
              onClick={e => trackFooterLink(e, linkbutton.content)}
            >
              {linkbutton.content}
            </a>
          </li>
        ))}
    </ul>
  </div>
);

export default LinkButton;
