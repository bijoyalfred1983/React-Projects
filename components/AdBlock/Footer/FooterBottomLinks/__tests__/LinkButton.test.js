/* @flow */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import LinkButton from '../LinkButton';
describe('<LinkButton />', () => {
  let PROPS;
  beforeEach(() => {
    PROPS = {
      bottomlinks:  [
        {
          id: '1',
          content: 'Terms & Conditions',
          url: '/termsCondition'
        },
        {
          id: '2',
          content: 'Privacy Policy',
          url: ''
        }
      ]
    };
  });

  test('renders on LinkButton correctly', () => {
    const wrapper = mount(
        <LinkButton {...PROPS} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


