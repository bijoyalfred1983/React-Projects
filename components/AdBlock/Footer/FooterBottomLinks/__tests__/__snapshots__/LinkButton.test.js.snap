// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> renders on LinkButton correctly 1`] = `
<LinkButton
  bottomlinks={
    Array [
      Object {
        "content": "Terms & Conditions",
        "id": "1",
        "url": "/termsCondition",
      },
      Object {
        "content": "Privacy Policy",
        "id": "2",
        "url": "",
      },
    ]
  }
>
  <div
    className="foooterBottomLinks"
  >
    <ul />
  </div>
</LinkButton>
`;
